services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -107,21 +106,24 @@ requestDevice (zusr ::: tid ::: uid ::: _) = do
     userLHStatus <- LegalHoldData.getUserLegalHoldStatus uid
     case userLHStatus of
         UserLegalHoldEnabled -> throwM userLegalHoldAlreadyEnabled
-        UserLegalHoldPending -> requestDeviceFromService
-        UserLegalHoldDisabled -> requestDeviceFromService
+        UserLegalHoldPending -> provisionLHDevice
+        UserLegalHoldDisabled -> provisionLHDevice
   where
-    requestDeviceFromService :: Galley Response
+    provisionLHDevice :: Galley Response
+    provisionLHDevice = do
+        (lastPrekey', prekeys) <- requestDeviceFromService
+        -- We don't distinguish the last key here; brig will do so when the device is added
+        LegalHoldData.insertPendingPrekeys uid (unpackLastPrekey lastPrekey' : prekeys)
+        LegalHoldData.setUserLegalHoldStatus uid UserLegalHoldPending
+        notifyClientsAboutLegalHoldRequest zusr uid lastPrekey' prekeys
```

Comment:
I'm assuming this only returns when brig has recorded the notifcation and it can't get lost any more?

Reply:
AFAICT yes the brig endpoint is synchronous.